packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart

```diff
@@ -348,12 +350,37 @@ represents the content of the collection must be in the same file.
         type.isDartCoreBool ||
         type.isPrimitiveList ||
         type.isJsonDocumentReference ||
+        type.isEnum ||
         dateTimeChecker.isAssignableFromType(type) ||
         timestampChecker.isAssignableFromType(type) ||
         geoPointChecker.isAssignableFromType(type);
     // TODO filter list other than List<string|bool|num>
   }
 
+  static ParameterMapping? _parameterMappingForType(DartType type) {
+    ParameterMapping enumMappingForType(DartType type) =>
+        (String name, bool nullable) {
+          final mapped =
+              '_\$${type.getDisplayString(withNullability: false)}EnumMap[$name]!';
+          if (nullable) {
+            return '$name == null ? null : $mapped';
+          } else {
+            return mapped;
+          }
+        };
+
+    if (type.isDartCoreList &&
```

Comment:
Why do we have list-specific logic?

Reply:
Because lists of enums need mapped to lists of strings

Reply:
How do you suggest we generate this without relying on list-specific logic?

<img width="571" alt="Screenshot 2023-07-31 at 9 22 33 AM" src="https://github.com/firebase/flutterfire/assets/7896519/3a478d0e-9de9-46c3-ab90-a37e19c56227">

Reply:
Maybe we could call the toJson method first and then pipe the resulting parameter in? That seems kind of jank though.

Reply:
Folks using enums should use json_serializable on their enums. Meaning they use things like `@JsonValue` & stuff, or maybe have a from/toJson on their enum class.


In fact, this _should_ work today. Let me double check

Reply:
Right, I got confused for a bit.
What needs updating is different. We need to rely on _$TestEnumPerFieldToJson, by asking users to enable `@JsonSeriaizable(createPerFieldToJson: true)`

That's something I added in json-serializable for this purpose https://github.com/google/json_serializable.dart/issues/972

Reply:
TL;DR the `update` should serialize fields as followed:

```dart
        _$EnumCollectionFieldMap['myField']!: _$EnumCollectionPerFieldToJson.myField(myField as MyFieldType),
```

This should add support for enums, but also `List<Enum>` or `Map<Enum, Whatever>` or whatever other funky combination you could think of.